File: backend/src/main/java/com/smithlife/backend/entity/Announcement.java
package com.smithlife.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "announcement")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Announcement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "announcement_id")
    private Long announcementId;

    @Column(name = "title", nullable = false, length = 200)
    private String title;

    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(name = "tag", nullable = false)
    private Tag tag;

    @Column(name = "image_url", length = 500)
    private String imageUrl;

    @Column(name = "is_new", nullable = false)
    private Boolean isNew;

    @Column(name = "published_at", nullable = false)
    private LocalDateTime publishedAt;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    public void update(String title, String content, Tag tag, String imageUrl) {
        this.title = title;
        this.content = content;
        this.tag = tag;
        this.imageUrl = imageUrl;
    }

    public void deactivate() {
        this.isActive = false;
    }

    public enum Tag {
        NOTICE, EVENT
    }
}

File: frontend/lib/data/models/announcement/announcement.dart
class Announcement {
  final int announcementId;
  final String title;
  final String content;
  final String tag; // 'NOTICE' | 'EVENT'
  final String? imageUrl;
  final bool? isNew;
  final String? publishedAt;
  final String? createdAt;

  Announcement({
    required this.announcementId,
    required this.title,
    required this.content,
    required this.tag,
    this.imageUrl,
    this.isNew,
    this.publishedAt,
    this.createdAt,
  });

  factory Announcement.fromJson(Map<String, dynamic> json) {
    return Announcement(
      announcementId: json['announcementId'],
      title: json['title'],
      content: json['content'],
      tag: json['tag'] ?? 'NOTICE',
      imageUrl: json['imageUrl'],
      isNew: json['isNew'],
      publishedAt: json['publishedAt']?.toString(),
      createdAt: json['createdAt']?.toString(),
    );
  }
}
